### ISSUE TITLE
Type narrowing on type[T] where T is a union alias of classes

### ISSUE THREAD

**Issue by LoicRiegel** [astral-sh/ty#3905]
### Summary

Hi! I hope this is not a duplicate:

`main.py`:

```py
from typing import assert_never

class A: ...
class B: ...

type T = A | B

def f(t: type[T]) -> str:
    if issubclass(t, A):
        return "A"
    if issubclass(t, B):
        return "B"

    assert_never(t)
```

Ty reports this:
```
Type `@Todo & ~type[A] & ~type[B]` is not equivalent to `Never` (type-assertion-failure) [Ln 15, Col 5]
Inferred type of argument is `@Todo & ~type[A] & ~type[B]` [Ln 15, Col 18]
info: `Never` and `@Todo & ~type[A] & ~type[B]` are not equivalent types
```

`ty.json`:

```json
{
    "environment": {
        "python-version": "3.12"
    },
    "rules": {
        "undefined-reveal": "ignore"
    }
}
```

[Playground](https://play.ty.dev/5caa6850-34e6-4dc2-8d2f-24901a6e54a3)

I ran the same thing with mypy and looks like it's supported there.

Note that the equivalent code using an explicit union works with ty:
```py

def f(t: type[A] | type[B]) -> str:
    if issubclass(t, A):
        return "A"
    if issubclass(t, B):
        return "B"

    assert_never(t)
```

### Version

ty 0.0.56 (3078f9fa2 2026-07-01)

**Comment by sharkdp** [comment]
Thank you for reporting this.

It looks like a Todo type for intersections in `type[..]`. I am removing this Todo in [my PR here](https://github.com/astral-sh/ruff/pull/26365), so this should hopefully soon be supported.

**Comment by sharkdp** [comment]
Btw, intersections are relevant here, because in the negated case, `~type[A | B]` turns into `~type[A] & ~type[B]` (de Morgan's law).

**Comment by carljm** [comment]
I think the real root cause here is simpler to fix: it's the PEP 695 type alias. Without that, the example works as expected. This Todo type just then propagates to the later intersection.

So this is a duplicate of #3996.
